### PR TITLE
don't release session after websocket closed

### PR DESCRIPTION
**Pull request by xmh19936688**
session maybe out-of-date after websocket closed

### CODE REVIEW

**Comment by Unknwon** [comment]
Hi, can you explain more? Why needs this fix, fixed what problem?

**Comment by xmh19936688** [comment]
1st: at 10:00AM, websocket request will block at `ctx.Next()` and `sess` fetched at the same time.
2nd: at 10:05AM, post request come in and updated session in `ctx.Next()` then invoke `sess.Release()` push data to redis.
3rd: at 10:07AM, websocket disconnect and `ctx.Next()` return, then session(fetched at 10:00AM) will invoke `sess.Releash()` and push old data to redis.

**Comment by Unknwon** [comment]
Thank for the use case!

But I think it is better to make this a config option of session, instead of changing the current default behavior.

**Comment by xmh19936688** [comment]
Like this?

File: session.go [hunk 1]
```diff
@@ -186,6 +191,10 @@ func Sessioner(options ...Options) macaron.Handler {
 
 		ctx.Next()
 
+		if ctx.Req.Header.Get("Upgrade") == "websocket" && manager.opt.IgnoreReleaseForWebSocket {
```
Comment (Unknwon) [inline session.go]: `manager.opt.IgnoreReleaseForWebSocket` should come before `ctx.Req.Header.Get("Upgrade") == "websocket"`.

**Comment by Unknwon** [comment]
Thanks for the update. One last mirror problem and we are good to go!

**Comment by Unknwon** [comment]
Thank you!

**Pull request merged**